opwen_email_server/utils/email_parser.py

```diff
@@ -80,3 +83,28 @@ def _get_recipients(email: dict) -> Iterable[str]:
 def get_domains(email: dict) -> Iterable[str]:
     return frozenset(address.split('@')[-1]
                      for address in _get_recipients(email))
+
+
+def _get_as_base64(image_url):
+    res = requests.get(image_url)
+
+    if res.status_code != 200:
+        res.raise_for_status()
+
+    img_type = res.headers['Content-Type']
+    img_content = b64encode(res.content).decode('ascii')
+    base64 = "data:" + img_type + ";base64," + img_content
+
+    return base64
+
+
+def convert_img_url_to_base64(email: dict) -> dict:
+    email_body = email.get('body', '')
+    if email_body:
```

Comment:
Given that all the logic should only be executed if there is an email body, it'll be neater to just early return here since that'll remove an indentation level for the normal code path.

```py
if not email.get('body'):
    return email

# processing logic goes here
```

Generally speaking, having the "error codepath" in indentation and the "happy codepath" without indentation is considered more readable. Here's an [article](http://codergears.com/Blog/?p=2392) about the use of early returns in the SQLite codebase that you could read if you're more interested in this.